cmd/traffic_cache_tool/CacheTool.cc

```diff
@@ -345,28 +354,47 @@ Stripe::updateHeaderFooter()
     zret.push(0, 1, "Writing Not Enabled.. Please use --write to enable writing to disk");
     return zret;
   }
-  for (int i = 0; i < 2; i++) {
-    for (int j = 0; j < 2; j++) {
-      char *meta_t = (char *)ats_memalign(ats_pagesize(), this->vol_dirlen());
-      memcpy(meta_t, &_meta[i][j], sizeof(StripeMeta));
-      CacheStoreBlocks hdr_size = round_up(sizeof(StripeMeta));
-      ssize_t n                 = pwrite(_span->_fd, meta_t, hdr_size, _meta_pos[i][j]);
-      if (n < hdr_size) {
-        std::cout << "problem writing to disk: " << strerror(errno) << ":"
-                  << " " << n << std::endl;
-        zret = Errata::Message(0, errno, "Failed to write stripe header ");
-        return zret;
-      }
+
+  char *meta_t = (char *)ats_memalign(ats_pagesize(), this->vol_dirlen());
+  // copy headers
+  for (auto i : {A, B}) {
+    // copy header
+    memcpy(meta_t, &_meta[i][HEAD], sizeof(StripeMeta));
+    // copy freelist
+    memcpy(meta_t + sizeof(StripeMeta) - sizeof(uint16_t), this->freelist, this->_segments * sizeof(uint16_t));
+
+    int64_t hdr_size = this->vol_headerlen();
+    ssize_t n        = pwrite(_span->_fd, meta_t, hdr_size, _meta_pos[i][HEAD]);
+    if (n < hdr_size) {
+      std::cout << "problem writing header to disk: " << strerror(errno) << ":"
+                << " " << n << "<" << hdr_size << std::endl;
+      zret = Errata::Message(0, errno, "Failed to write stripe header ");
+      return zret;
+    }
+    // copy dir entries
+    int64_t dir_size = this->vol_dirlen() - vol_headerlen() - ROUND_TO_STORE_BLOCK(sizeof(StripeMeta));
```

Comment:
I'd move this up to above the allocate and initialize it to `vol_dirlen()`. Then you can decrement it here to adjust.